src/System.Text.Json/src/System/Text/Json/Node/JsonObject.cs

```diff
@@ -0,0 +1,465 @@
+        public void Add(string propertyName, decimal propertyValue) => Add(propertyName, new JsonNumber(propertyValue));
+
+        /// <summary>
+        ///   Adds the properties from the specified collection to the JSON object.
+        /// </summary>
+        /// <param name="jsonProperties">Properties to add.</param>
+        /// <exception cref="ArgumentException">
+        ///   Provided collection contains duplicates if handling duplicates is set to <see cref="DuplicatePropertyNameHandling.Error"/>.
+        /// </exception>
+        /// <exception cref="ArgumentNullException">
+        ///   Some of property names are null.
+        /// </exception>
+        public void AddRange(IEnumerable<KeyValuePair<string, JsonNode>> jsonProperties)
+        {
+            foreach (KeyValuePair<string, JsonNode> property in jsonProperties)
+            {
+                Add(property);
+            }
+        }
+
+        /// <summary>
+        ///   Removes the property with the specified name.
+        /// </summary>
+        /// <param name="propertyName"></param>
+        /// <returns>
+        ///   <see langword="true"/> if the property is successfully found in a JSON object and removed,
+        ///   <see langword="false"/> otherwise.
+        /// </returns>
+        /// <exception cref="ArgumentNullException">
+        ///   Provided property name is null.
+        /// </exception>
+        public bool Remove(string propertyName) => propertyName != null ? _dictionary.Remove(propertyName) : throw new ArgumentNullException(nameof(propertyName));
+
+        /// <summary>
+        ///   Determines whether a property is in a JSON object.
+        /// </summary>
+        /// <param name="propertyName">Name of the property to check.</param>
+        /// <returns>
+        ///   <see langword="true"/> if the property is successfully found in a JSON object,
+        ///   <see langword="false"/> otherwise.
+        /// </returns>
+        /// <exception cref="ArgumentNullException">
+        ///   Provided property name is null.
+        /// </exception>
+        public bool ContainsProperty(string propertyName) => propertyName != null ? _dictionary.ContainsKey(propertyName) : throw new ArgumentNullException(nameof(propertyName));
+
+        /// <summary>
+        ///   Modifies the name of specified property.
+        /// </summary>
+        /// <param name="oldName">Old name of the property to change.</param>
+        /// <param name="newName">New name of the property to change.</param>
+        /// <exception cref="ArgumentException">
+        ///   Property name to set already exists if handling duplicates is set to <see cref="DuplicatePropertyNameHandling.Error"/>.
+        /// </exception>
+        /// <exception cref="KeyNotFoundException">
+        ///   Property with specified name is not found in JSON object.
+        /// </exception>
+        /// <remnark>
+        ///   Does not guarantee keeping the order the same.
```

Comment:
NIT: `Does not guarantee keeping the order the same.` -> `Does not guarantee keeping the same order.`